src/GitHubTools/Functions/Public/Update-AzurePipelineDependencies.ps1

```diff
@@ -0,0 +1,93 @@
+<#
+
+.SYNOPSIS
+Updates the refs of the azure-pipelines.yml file's repository resources to the latest
+
+.DESCRIPTION
+Updates the refs of the azure-pipelines.yml file's repository resources to the latest
+
+.PARAMETER GitHubOrganisation
+The GitHub organisation
+
+.PARAMETER RepositoryPrefix
+The prefix of repositories to create new branches with updated pipeline dependencies
+
+.EXAMPLE
+Set-GitHubSessionInformation -PatToken <not-a-real-pat-token>
+Update-AzurePipelineDependencies -GitHubOrganisation MyOrganisation -RepositoryPrefix foo-
+
+.NOTES
+powershell-yaml powershell module is required. Install-Module -name powershell-yaml
+The powershell-yaml module is no longer being maintained https://github.com/aaubry/YamlDotNet/discussions/689
+
+#>
+function Update-AzurePipelineDependencies {
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSReviewUnusedParameter", "RepositoryPrefix", Justification = "False positive as rule does not know that Where-Object operates within the same scope")]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseDeclaredVarsMoreThanAssignments", "UpdatePipelineYml", Justification="False positive as rule does not know that variable is updated in ForEach-Object")]
```

Comment:
If you use `foreach` rather than `For-EachObject` you won't need to use this suppression.  Is there a reason `For-EachObject` is preferred in this case?

Reply:
Good shout, have changed to use foreach